Content.Shared/Damage/Components/DamageableComponent.cs

```diff
@@ -374,6 +387,26 @@ private void OnHealthChanged(List<DamageChangeData> changes)
             OnHealthChanged(args);
         }
 
+        private IReadOnlyDictionary<DamageGroupPrototype, int> damageListToDamageGroup(IReadOnlyDictionary<DamageTypePrototype, int> damagelist)
+        {
+            var damageGroupDict = new Dictionary<DamageGroupPrototype, int>();
+            int damageGroupSumDamage = 0;
+            int damageTypeDamage = 0 ;
+            foreach (var damageGroup in SupportedGroups)
+            {
+                damageGroupSumDamage = 0;
+                foreach (var damageType in SupportedTypes)
+                {
+                    damageTypeDamage = 0;
+                     damagelist.TryGetValue(damageType,out damageTypeDamage);
+                     damageGroupSumDamage += damageTypeDamage;
+                }
+                damageGroupDict.Add(damageGroup,damageGroupSumDamage);
+            }
+
+            return damageGroupDict;
+        }
```

Comment:
I have to say it is less pleasant when easily visible.
The method name should be capitalized though, DamageListToDamageGroup.